### PR TITLE
Fix #1892: Feature to show/hide password on login, signup, change & reset password pages

### LINKED ISSUE #1892: Add support for show and hide password
### Observed Behaviour
Currently , there is no feature for showing or hiding password in login and sign up page.

### Expected Behaviour
There should be a one eye icon which on clicking shows or hides the password.

### PR DESCRIPTION
**Pull request by Sanji515**
Fixes #1892 

Added feature for showing or hiding the password in-

- Login and SignUp
- change-password under the profile tab
- forgot-password in Login & Signup Page

### CODE REVIEW

**Comment by Sanji515** [comment]
@gautamjajoo @RishabhJain2018 I've done the same for `change-password` under the Update profile tab and also for the `forgot-password` in Login & Signup Page

**Comment by gautamjajoo** [comment]
@Sanji515 Can you share a screencast of it?

**Comment by Sanji515** [comment]
@gautamjajoo @RishabhJain2018  Here's the screenshot-- `change-password` in Update Profile and `forgot-password` in Login & Signup Page.

Update Profile =>

screenshot 146

Forgot Password =>

screenshot 147

Login & Signup Page =>

screenshot 150

**Comment by Sanji515** [comment]
@gautamjajoo @RishabhJain2018 Here's the screencast.

togglepasswordvisibility

**Comment by Sanji515** [comment]
@gautamjajoo @RishabhJain2018 Please review

**Comment by RishabhJain2018** [comment]
@galipremsagar Can you please review this PR?

File: frontend/src/js/controllers/authCtrl.js [hunk 1]
```diff
@@ -36,6 +36,10 @@
         vm.isValid = {};
         vm.confirmMsg = '';
         $rootScope.loaderTitle = '';
+        $rootScope.passwordInputType = 'password';
```
Comment (galipremsagar) [inline frontend/src/js/controllers/authCtrl.js]:
```suggestion
        $rootScope.passwordInputType = 'password';
```

File: frontend/src/js/controllers/authCtrl.js [hunk 2]
```diff
@@ -36,6 +36,10 @@
...
+        $rootScope.confirmPasswordInputType = 'password';
```
Comment (galipremsagar) [inline frontend/src/js/controllers/authCtrl.js]:
```suggestion
        $rootScope.canShowConfirmPassword = false;
```

File: frontend/src/js/controllers/authCtrl.js [hunk 3]
```diff
@@ -36,6 +36,10 @@
...
+        $rootScope.showEyeIcon = false;
```
Comment (galipremsagar) [inline frontend/src/js/controllers/authCtrl.js]: You might not need showEyeIcon anymore as you can toggle using the canShowPassword variable.

File: frontend/src/views/web/auth/login.html [hunk 4]
```diff
@@ -20,8 +20,11 @@
         </div>
         <!-- password -->
         <div class="input-field align-left">
-            <input type="password" id="password" onpaste="return false;" class="dark-autofill" name="password" ng-model="auth.getUser.password" autocomplete="new-password" ng-minlength="8" required>
-            <span class="form-icon form-icon-dark"><i class="fa fa-key"></i></span>
+            <input type="{{passwordInputType}}" id="password" onpaste="return false;" class="dark-autofill" name="password" ng-model="auth.getUser.password" autocomplete="new-password" ng-minlength="8" required>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/login.html]:
```suggestion
            <input type="{{canShowPassword ? 'text' : 'password'}}" id="password" onpaste="return false;" class="dark-autofill" name="password" ng-model="auth.getUser.password" autocomplete="new-password" ng-minlength="8" required>
```

File: frontend/src/views/web/auth/reset-password-confirm.html [hunk 5]
```diff
@@ -19,17 +19,23 @@
         <form name="resetconfirmForm" ng-submit="auth.resetPasswordConfirm(resetconfirmForm.$valid)" class="reset-password" novalidate>
             <!-- email -->
             <div class="input-field align-left">
-                <input type="password" id="new_password1" class="dark-autofill" name="new_password1" ng-model="auth.getUser.new_password1" minlength="8" required>
-                <span class="form-icon form-icon"><i class="fa fa-key"></i></span>
+                <input type="{{passwordInputType}}" id="new_password1" onpaste="return false;" class="dark-autofill" name="new_password1" ng-model="auth.getUser.new_password1" minlength="8" required>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/reset-password-confirm.html]:
```suggestion
                <input type="{{canShowPassword ? 'text' : 'password'}}" id="new_password1" onpaste="return false;" class="dark-autofill" name="new_password1" ng-model="auth.getUser.new_password1" minlength="8" required>
```

File: frontend/src/views/web/auth/reset-password-confirm.html [hunk 6]
```diff
@@ -19,17 +19,23 @@
...
                 <label for="new_password1">New Password (min 8 characters) *</label>
                 <div class="wrn-msg text-highlight" ng-messages="resetconfirmForm.new_password1.$error" ng-if="resetconfirmForm.new_password1.$touched || resetconfirmForm.$submitted">
                     <p ng-message="minlength">Password is less than 8 characters.</p>
                     <p ng-message="required">Password is required.</p>
                 </div>
             </div>
             <div class="input-field align-left">
-                <input type="password" id="new_password2" class="dark-autofill" name="new_password2" ng-model="auth.getUser.new_password2" minlength="8" compare-to="auth.getUser.new_password1" required>
-                <span class="form-icon form-icon"><i class="fa fa-key"></i></span>
+                <input type="{{confirmPasswordInputType}}" id="new_password2" class="dark-autofill" name="new_password2" ng-model="auth.getUser.new_password2" minlength="8" compare-to="auth.getUser.new_password1" required>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/reset-password-confirm.html]:
```suggestion
                <input type="{{canShowConfirmPassword ? 'text' : 'password'}}" id="new_password2" class="dark-autofill" name="new_password2" ng-model="auth.getUser.new_password2" minlength="8" compare-to="auth.getUser.new_password1" required>
```

File: frontend/src/views/web/auth/signup.html [hunk 7]
```diff
@@ -30,8 +30,11 @@
         </div>
         <!-- password -->
         <div class="input-field align-left">
-            <input type="password" id="password" onpaste="return false;" class="dark-autofill" name="password" ng-model="auth.regUser.password" ng-minlength="8" ng-change="auth.checkStrength(auth.regUser.password)" required>
-            <span class="form-icon form-icon-dark"><i class="fa fa-key"></i></span>
+            <input type="{{passwordInputType}}" id="password" onpaste="return false;" class="dark-autofill" name="password" ng-model="auth.regUser.password" ng-minlength="8" ng-change="auth.checkStrength(auth.regUser.password)" required>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/signup.html]:
```suggestion
            <input type="{{canShowPassword ? 'text' : 'password'}}" id="password" onpaste="return false;" class="dark-autofill" name="password" ng-model="auth.regUser.password" ng-minlength="8" ng-change="auth.checkStrength(auth.regUser.password)" required>
```

File: frontend/src/views/web/auth/signup.html [hunk 8]
```diff
@@ -43,8 +46,11 @@
         </div>
         <!-- confirm password -->
         <div class="input-field align-left">
-            <input type="password" id="confirm_password" onpaste="return false;" class="dark-autofill" name="confirm_password" ng-model="auth.regUser.confirm_password" ng-minlength="8" compare-to="auth.regUser.password" required>
-            <span class="form-icon form-icon-dark"><i class="fa fa-check-circle"></i></span>
+            <input type="{{confirmPasswordInputType}}" id="confirm_password" onpaste="return false;" class="dark-autofill" name="confirm_password" ng-model="auth.regUser.confirm_password" ng-minlength="8" compare-to="auth.regUser.password" required>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/signup.html]:
```suggestion
            <input type="{{canShowPassword ? 'text' : 'password'}}" id="confirm_password" onpaste="return false;" class="dark-autofill" name="confirm_password" ng-model="auth.regUser.confirm_password" ng-minlength="8" compare-to="auth.regUser.password" required>
```

File: frontend/src/views/web/change-password.html [hunk 9]
```diff
@@ -6,17 +6,23 @@
                 <div class="pass-title">Change Password</div>
                 <form name="changepwdForm" ng-submit="changepwd.changePassword(changepwdForm.$valid)" novalidate>
                     <div class="input-field align-left">
-                        <input id="old_password" name="old_password" type="password" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.old_password" ng-minlength="8" required focus-if>
-                        <span class="form-icon form-icon-dark"><i class="fa fa-key"></i></span>
+                        <input id="old_password" name="old_password" type="{{oldpwdInputType}}" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.old_password" ng-minlength="8" required focus-if>
```
Comment (galipremsagar) [inline frontend/src/views/web/change-password.html]:
```suggestion
                        <input id="old_password" name="old_password" type="{{canShowOldPassword ? 'text' : 'password'}}" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.old_password" ng-minlength="8" required focus-if>
```

File: frontend/src/views/web/change-password.html [hunk 10]
```diff
@@ -6,17 +6,23 @@
...
                         <label for="old_password">Old Password*</label>
                         <div class="wrn-msg text-highlight" ng-messages="changepwdForm.old_password.$error" ng-if="changepwdForm.old_password.$touched || changepwdForm.$submitted">
                             <p ng-message="minlength">password is less than 8 characters.</p>
                             <p ng-message="required">Old password is required</p>
                         </div>
                     </div>
                     <div class="input-field align-left">
-                        <input id="new_password1" name="new_password1" type="password" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.new_password1" ng-minlength="8" required match="changepwd.user.old_password">
-                        <span class="form-icon form-icon-dark"><i class="fa fa-key"></i></span>
+                        <input id="new_password1" name="new_password1" type="{{newpwdInputType}}" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.new_password1" ng-minlength="8" required match="changepwd.user.old_password">
```
Comment (galipremsagar) [inline frontend/src/views/web/change-password.html]:
```suggestion
                        <input id="new_password1" name="new_password1" type="{{canShowNewPassword ? 'text' : 'password'}}" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.new_password1" ng-minlength="8" required match="changepwd.user.old_password">
```

File: frontend/src/views/web/change-password.html [hunk 11]
```diff
@@ -25,8 +31,11 @@
                         </div>
                     </div>
                     <div class="input-field align-left">
-                        <input id="new_password2" name="new_password2" type="password" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.new_password2" ng-minlength="8" compare-to="changepwd.user.new_password1" required>
-                        <span class="form-icon form-icon-dark"><i class="fa fa-key"></i></span>
+                        <input id="new_password2" name="new_password2" type="{{newpwdAgainInputType}}" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.new_password2" ng-minlength="8" compare-to="changepwd.user.new_password1" required>
```
Comment (galipremsagar) [inline frontend/src/views/web/change-password.html]:
```suggestion
                        <input id="new_password2" name="new_password2" type="{{canShowNewConfirmPassword ? 'text' : 'password'}}" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.new_password2" ng-minlength="8" compare-to="changepwd.user.new_password1" required>
```

File: frontend/src/views/web/auth/login.html [hunk 12]
```diff
@@ -20,8 +20,11 @@
...
+            <span class="form-icon form-icon-dark" ng-click="auth.togglePasswordVisibility()">
+                <i ng-if="!showEyeIcon" class="fa fa-eye"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/login.html]:
```suggestion
                <i ng-if="!canShowPassword" class="fa fa-eye"></i>
```

File: frontend/src/views/web/auth/login.html [hunk 13]
```diff
@@ -20,8 +20,11 @@
...
+                <i ng-if="showEyeIcon" class="fa fa-eye-slash"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/login.html]:
```suggestion
                <i ng-if="canShowPassword" class="fa fa-eye-slash"></i>
```

File: frontend/src/views/web/auth/reset-password-confirm.html [hunk 14]
```diff
@@ -19,17 +19,23 @@
         <form name="resetconfirmForm" ng-submit="auth.resetPasswordConfirm(resetconfirmForm.$valid)" class="reset-password" novalidate>
             <!-- email -->
             <div class="input-field align-left">
-                <input type="password" id="new_password1" class="dark-autofill" name="new_password1" ng-model="auth.getUser.new_password1" minlength="8" required>
-                <span class="form-icon form-icon"><i class="fa fa-key"></i></span>
+                <input type="{{passwordInputType}}" id="new_password1" onpaste="return false;" class="dark-autofill" name="new_password1" ng-model="auth.getUser.new_password1" minlength="8" required>
+                <span class="form-icon form-icon-dark" ng-click="auth.togglePasswordVisibility()">
+                    <i ng-if="!showEyeIcon" class="fa fa-eye"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/reset-password-confirm.html]:
```suggestion
                    <i ng-if="!canShowPassword" class="fa fa-eye"></i>
```

File: frontend/src/views/web/auth/reset-password-confirm.html [hunk 15]
```diff
@@ -19,17 +19,23 @@
...
+                    <i ng-if="showEyeIcon" class="fa fa-eye-slash"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/reset-password-confirm.html]:
```suggestion
                    <i ng-if="canShowPassword" class="fa fa-eye-slash"></i>
```

File: frontend/src/views/web/auth/reset-password-confirm.html [hunk 16]
```diff
@@ -19,17 +19,23 @@
...
                     <p ng-message="minlength">Password is less than 8 characters.</p>
                     <p ng-message="required">Password is required.</p>
                 </div>
             </div>
             <div class="input-field align-left">
-                <input type="password" id="new_password2" class="dark-autofill" name="new_password2" ng-model="auth.getUser.new_password2" minlength="8" compare-to="auth.getUser.new_password1" required>
-                <span class="form-icon form-icon"><i class="fa fa-key"></i></span>
+                <input type="{{confirmPasswordInputType}}" id="new_password2" class="dark-autofill" name="new_password2" ng-model="auth.getUser.new_password2" minlength="8" compare-to="auth.getUser.new_password1" required>
+                <span class="form-icon form-icon-dark" ng-click="auth.toggleConfirmPasswordVisibility()">
+                    <i ng-if="!showEyeIconConfirmpwd" class="fa fa-eye"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/reset-password-confirm.html]:
```suggestion
                    <i ng-if="!canShowConfirmPassword" class="fa fa-eye"></i>
```

File: frontend/src/views/web/auth/reset-password-confirm.html [hunk 17]
```diff
@@ -19,17 +19,23 @@
...
+                    <i ng-if="showEyeIconConfirmpwd" class="fa fa-eye-slash"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/reset-password-confirm.html]:
```suggestion
                    <i ng-if="canShowConfirmPassword" class="fa fa-eye-slash"></i>
```

File: frontend/src/views/web/auth/signup.html [hunk 18]
```diff
@@ -30,8 +30,11 @@
         </div>
         <!-- password -->
         <div class="input-field align-left">
-            <input type="password" id="password" onpaste="return false;" class="dark-autofill" name="password" ng-model="auth.regUser.password" ng-minlength="8" ng-change="auth.checkStrength(auth.regUser.password)" required>
-            <span class="form-icon form-icon-dark"><i class="fa fa-key"></i></span>
+            <input type="{{passwordInputType}}" id="password" onpaste="return false;" class="dark-autofill" name="password" ng-model="auth.regUser.password" ng-minlength="8" ng-change="auth.checkStrength(auth.regUser.password)" required>
+            <span class="form-icon form-icon-dark" ng-click="auth.togglePasswordVisibility()">
+                <i ng-if="!showEyeIcon" class="fa fa-eye"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/signup.html]:
```suggestion
                <i ng-if="!canShowPassword" class="fa fa-eye"></i>
```

File: frontend/src/views/web/auth/signup.html [hunk 19]
```diff
@@ -30,8 +30,11 @@
...
+                <i ng-if="showEyeIcon" class="fa fa-eye-slash"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/signup.html]:
```suggestion
                <i ng-if="canShowPassword" class="fa fa-eye-slash"></i>
```

File: frontend/src/views/web/auth/signup.html [hunk 20]
```diff
@@ -43,8 +46,11 @@
         </div>
         <!-- confirm password -->
         <div class="input-field align-left">
-            <input type="password" id="confirm_password" onpaste="return false;" class="dark-autofill" name="confirm_password" ng-model="auth.regUser.confirm_password" ng-minlength="8" compare-to="auth.regUser.password" required>
-            <span class="form-icon form-icon-dark"><i class="fa fa-check-circle"></i></span>
+            <input type="{{confirmPasswordInputType}}" id="confirm_password" onpaste="return false;" class="dark-autofill" name="confirm_password" ng-model="auth.regUser.confirm_password" ng-minlength="8" compare-to="auth.regUser.password" required>
+            <span class="form-icon form-icon-dark" ng-click="auth.toggleConfirmPasswordVisibility()">
+                <i ng-if="!showEyeIconConfirmpwd" class="fa fa-eye"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/signup.html]:
```suggestion
                <i ng-if="!canShowConfirmPassword" class="fa fa-eye"></i>
```

File: frontend/src/views/web/auth/signup.html [hunk 21]
```diff
@@ -43,8 +46,11 @@
...
+                <i ng-if="showEyeIconConfirmpwd" class="fa fa-eye-slash"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/auth/signup.html]:
```suggestion
                <i ng-if="canShowConfirmPassword" class="fa fa-eye-slash"></i>
```

File: frontend/src/views/web/change-password.html [hunk 22]
```diff
@@ -6,17 +6,23 @@
                 <div class="pass-title">Change Password</div>
                 <form name="changepwdForm" ng-submit="changepwd.changePassword(changepwdForm.$valid)" novalidate>
                     <div class="input-field align-left">
-                        <input id="old_password" name="old_password" type="password" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.old_password" ng-minlength="8" required focus-if>
-                        <span class="form-icon form-icon-dark"><i class="fa fa-key"></i></span>
+                        <input id="old_password" name="old_password" type="{{oldpwdInputType}}" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.old_password" ng-minlength="8" required focus-if>
+                        <span class="form-icon form-icon-dark" ng-click="changepwd.toggleOldpwdVisibility()">
+                            <i ng-if="!showEyeIconOldpwd" class="fa fa-eye"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/change-password.html]:
```suggestion
                            <i ng-if="!canShowOldPassword" class="fa fa-eye"></i>
```

File: frontend/src/views/web/change-password.html [hunk 23]
```diff
@@ -6,17 +6,23 @@
...
+                            <i ng-if="showEyeIconOldpwd" class="fa fa-eye-slash"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/change-password.html]:
```suggestion
                            <i ng-if="canShowOldPassword" class="fa fa-eye-slash"></i>
```

File: frontend/src/views/web/change-password.html [hunk 24]
```diff
@@ -6,17 +6,23 @@
...
                             <p ng-message="minlength">password is less than 8 characters.</p>
                             <p ng-message="required">Old password is required</p>
                         </div>
                     </div>
                     <div class="input-field align-left">
-                        <input id="new_password1" name="new_password1" type="password" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.new_password1" ng-minlength="8" required match="changepwd.user.old_password">
-                        <span class="form-icon form-icon-dark"><i class="fa fa-key"></i></span>
+                        <input id="new_password1" name="new_password1" type="{{newpwdInputType}}" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.new_password1" ng-minlength="8" required match="changepwd.user.old_password">
+                        <span class="form-icon form-icon-dark" ng-click="changepwd.toggleNewpwdVisibility()">
+                            <i ng-if="!showEyeIconNewpwd" class="fa fa-eye"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/change-password.html]:
```suggestion
                            <i ng-if="!canShowNewPassword" class="fa fa-eye"></i>
```

File: frontend/src/views/web/change-password.html [hunk 25]
```diff
@@ -6,17 +6,23 @@
...
+                            <i ng-if="showEyeIconNewpwd" class="fa fa-eye-slash"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/change-password.html]:
```suggestion
                            <i ng-if="canShowNewPassword" class="fa fa-eye-slash"></i>
```

File: frontend/src/views/web/change-password.html [hunk 26]
```diff
@@ -25,8 +31,11 @@
                         </div>
                     </div>
                     <div class="input-field align-left">
-                        <input id="new_password2" name="new_password2" type="password" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.new_password2" ng-minlength="8" compare-to="changepwd.user.new_password1" required>
-                        <span class="form-icon form-icon-dark"><i class="fa fa-key"></i></span>
+                        <input id="new_password2" name="new_password2" type="{{newpwdAgainInputType}}" class="text-dark-black dark-autofill w-400" ng-model="changepwd.user.new_password2" ng-minlength="8" compare-to="changepwd.user.new_password1" required>
+                        <span class="form-icon form-icon-dark" ng-click="changepwd.toggleNewpwdAgainVisibility()">
+                            <i ng-if="!showEyeIconNewpwdAgain" class="fa fa-eye"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/change-password.html]:
```suggestion
                            <i ng-if="!canShowNewConfirmPassword" class="fa fa-eye"></i>
```

File: frontend/src/views/web/change-password.html [hunk 27]
```diff
@@ -25,8 +31,11 @@
...
+                            <i ng-if="showEyeIconNewpwdAgain" class="fa fa-eye-slash"></i>
```
Comment (galipremsagar) [inline frontend/src/views/web/change-password.html]:
```suggestion
                            <i ng-if="canShowNewConfirmPassword" class="fa fa-eye-slash"></i>
```

File: frontend/src/js/controllers/authCtrl.js [hunk 28]
```diff
@@ -52,6 +56,28 @@
             vm.loginContainer.removeClass('low-screen');
         };
 
+        // toggle password visibility
+        vm.togglePasswordVisibility = function() {
```
Comment (galipremsagar) [inline frontend/src/js/controllers/authCtrl.js]: In all the toggle functions just do this for toggling their respective variables:
$scope.canShowPassword = !$scope.canShowPassword;

Please clean up all the newly added js code(toggle functions and their associated variables).

File: frontend/src/js/controllers/authCtrl.js (hunk 3 above)
Comment (Sanji515) [reply frontend/src/js/controllers/authCtrl.js]: done.

File: frontend/src/js/controllers/authCtrl.js (hunk 28 above)
Comment (Sanji515) [reply frontend/src/js/controllers/authCtrl.js]: @galipremsagar done

**Comment by Sanji515** [comment]
@galipremsagar Thanks for your review.
Made the suggested changes, please take a look, thanks!

**Comment by galipremsagar** [comment]
> @galipremsagar Thanks for your review.
> Made the suggested changes, please take a look, thanks!

Well done! 🎉  
@RishabhJain2018 this can be merged.

**Comment by RishabhJain2018** [comment]
@Sanji515 The cursor should change to pointer here: https://github.com/Cloud-CV/EvalAI/pull/2072#issuecomment-461000451
Can you add that change also?

**Comment by Sanji515** [comment]
> @Sanji515 The cursor should change to pointer here: [#2072 (comment)](https://github.com/Cloud-CV/EvalAI/pull/2072#issuecomment-461000451)
> Can you please add that change also?

@RishabhJain2018 Done. Please take a look.
thanks!

**Comment by RishabhJain2018** [comment]
Thanks @galipremsagar for the review.

**Pull request merged**